code.js

```diff
@@ -1,3 +1,38 @@
 function permutationSort(a) {
-    return 0;
+    function sorted(arr) {
+        for (var i = 0; i < arr.length - 1; i++) {
+            if (arr[i] > arr[i + 1]) {
+                return false;
+            }
+        }
+        return true;
+    }
+    
+    function swap(arr, i, j) {
+        const x = arr[i];
+        arr[i] = arr[j];
+        arr[j] = x;
+    }
+    
+    function permute(arr, leftInd) {
+        var perms = 0;
+        const l = leftInd + 1;
+        if(arr.length <= l){
+            return 1;
+        }
+        for (var i = leftInd; i < arr.length; i++) {
+            if (arr[leftInd] != arr[i]){
+                swap(arr, leftInd, i);
+            }
+            perms += permute(arr, l);
+            if (sorted(arr)){
+                return perms;
+            } else if (arr[leftInd] != arr[i]){
+                swap (arr, leftInd, i);
```

Comment:
Don't understand why you're doing this here -- isn't this the same as you're doing above?

Reply:
It is undoing the first swap so that the array is in its original order so it can move on to the next permutation. After more testing, it appears that the "else if" is unnecessary and it functions the same without it.